### PR TITLE
Feature/upgrade ethereumjs util

### PR DESCRIPTION
**Pull request by rickycodes**
Upgrade `ethereumjs-util` to `latest`

Most of the changes are related to the fact that the addresses we used previously are now considered invalid (they were always invalid, but `ethUtil` now throws in cases where they are).

### CODE REVIEW

**Comment by brad-decker** [comment]
https://github.com/MetaMask/controllers/blob/f421f1ed1b5e70a4a2a224569e49f77e8bc0b53a/src/util.ts#L284 this line would previously return false for input such as '0000000000000000000000000000000000000000' (non prefixed hex string) and now will throw an error. I highly, highly, highly recommend not using this method directly and instead use a helper method that does something like this one: 

https://github.com/MetaMask/metamask-extension/pull/11089/files#diff-f6b120c1869bc7363902d972ebbc6b34fef85122aa58b606f5fdddec01dfb049R14-R31

which I created to get around the numerous breaking changes this introduced and going on our third hotfix for 9.5.3

File: src/keyring/KeyringController.ts [hunk 1]
```diff
@@ -561,9 +566,7 @@ export class KeyringController extends BaseController<
         async (keyring: KeyringObject, index: number): Promise<Keyring> => {
           const keyringAccounts = await keyring.getAccounts();
           const accounts = Array.isArray(keyringAccounts)
-            ? keyringAccounts.map((address) =>
-                ethUtil.toChecksumAddress(address),
-              )
+            ? keyringAccounts.map((address) => toChecksumAddress(address))
```
Comment (brad-decker) [inline src/keyring/KeyringController.ts:569]: is it possible that keyringAccounts could ever return non `0x` prefixed hex strings?

File: src/user/AddressBookController.ts [hunk 2]
```diff
@@ -118,7 +118,7 @@ export class AddressBookController extends BaseController<
    */
   set(address: string, name: string, chainId = '1', memo = '') {
     address = toChecksumAddress(address);
-    if (!isValidAddress(address)) {
+    if (!isValidHexAddress(address)) {
```
Comment (brad-decker) [inline src/user/AddressBookController.ts:121]: the line above this where we call toChecksumAddress will fail if dealing with a non 0x prefixed hex string or any other garbage input. I think you could just check if it's a valid Hex address first. Also not a crazy big fan of reassigning a function argument but that's not on you lol.

**Review by brad-decker — CHANGES_REQUESTED** [review]

File: src/keyring/KeyringController.ts (hunk 1 above)
Comment (rickycodes) [reply src/keyring/KeyringController.ts]: I've added `toChecksumHexAddress` as discussed and used that here now.

File: src/user/AddressBookController.ts (hunk 2 above)
Comment (rickycodes) [reply src/user/AddressBookController.ts:124]: I've added `toChecksumHexAddress` and used it in place of `toChecksumAddress` for safety++

**Review by brad-decker — APPROVED** [review]
LGTM!

**Review by brad-decker — APPROVED** [review]

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**